x/gamm/simulation/operations.go

```diff
@@ -204,17 +213,25 @@ func SimulateMsgSwapExactAmountIn(ak stakingTypes.AccountKeeper, bk stakingTypes
 func RandomExactAmountInRoute(ctx sdk.Context, r *rand.Rand, k keeper.Keeper, tokenIn sdk.Coin) (res []types.SwapAmountInRoute, tokenOut sdk.Coin) {
 	routeLen := r.Intn(1) + 1
 
-	pools, err := k.GetPools(ctx)
+	allpools, err := k.GetPools(ctx)
 	if err != nil {
 		panic(err)
 	}
+
+	pools := []types.PoolI{}
+	for _, pool := range allpools {
+		if pool.IsActive(ctx.BlockTime()) {
+			pools = append(pools, pool)
+		}
+	}
+
 	if len(pools) == 0 {
 		return
 	}
 
-	res = make([]types.SwapAmountInRoute, routeLen)
-	for i := range res {
-		for {
+	res = []types.SwapAmountInRoute{}
+	for i := 0; i < routeLen; i++ {
+		for retry := 0; retry < 10; retry++ {
```

Comment:
Whats this retry for?

Reply:
We need to find a pool that contains the input coin as an asset, but the randomly selected pool may not.